src/guides/v2.2/coding-standards/docblock-standard-general.md

```diff
@@ -444,31 +444,28 @@ For example:
 /**
  * Recursively delete directory from storage
  *
- * @param  string $path Target dir
+ * @param string $path Absolute path to target directory
  * @return void
- * @throws Mage_Core_Exception when directories cannot be deleted
+ * @throws \Magento\Framework\Exception\LocalizedException
  */
 public function deleteDirectory($path)
 {
-    // prevent accidental root directory deleting
-    $rootCmp = rtrim($this->getHelper()->getStorageRoot(), DS);
-    $pathCmp = rtrim($path, DS);
-
-    if ($rootCmp == $pathCmp) {
-        Mage::throwException(Mage::helper('Mage_Cms_Helper_Data')->__('Cannot delete root directory %s.', $path));
+    if ($this->_coreFileStorageDb->checkDbUsage()) {
+        $this->_directoryDatabaseFactory->create()->deleteDirectory($path);
     }
-
-    $io = new Varien_Io_File();
-
-    if (Mage::helper('Mage_Core_Helper_File_Storage_Database')->checkDbUsage()) {
-        Mage::getModel('Mage_Core_Model_File_Storage_Directory_Database')->deleteDirectory($path);
+    if (!$this->isPathAllowed($path, $this->getConditionsForExcludeDirs())) {
+        throw new \Magento\Framework\Exception\LocalizedException(
```

Comment:
This is explicit throwing of an exception. The section is about implicit throwing. An example could be if there would be a method like `$this->throwException()` or something like this.

Reply:
I have change the example. Please check it. Thanks!